src/shared/getFunction.js

```diff
@@ -0,0 +1,10 @@
+export let functions = {};
+
+export default function getFunction ( str, i ) {
+	if ( functions[ str ] ) return functions[ str ];
+
+	let args = new Array( i );
+	while ( i-- ) args[i] = `_${i}`;
+
+	return ( functions[ str ] = new Function( args.join( ',' ), `return (${str})` ) );
```

Comment:
@martypdx no need to move anything, if functions are in the template, we don't get past 4th line. New function is created only if there isn't one (parsing without `insertExpressions`).